### ISSUE TITLE
Idea for spawn 

### ISSUE THREAD

**Issue by liexes** [diamond-rip/Eden#432]
The idea is a free spawn, instead of freezed The player can roam freely on the map until the start counter ends, when Ester finishes, it will return him to the spawn position to start the game.

**Comment by RainFallDev** [comment]
Make sure use this version https://drive.google.com/file/d/1fSzljrVbm2jEiCx121aKyOc9JE1p4ESq/view?usp=sharing

**Comment by RainFallDev** [comment]
https://github.com/diamond-rip/Eden/issues/431

**Comment by liexes** [comment]
Yes, but what does that have to do with my suggestion?

**Comment by liexes** [comment]
Is this option already added?

**Comment by RainFallDev** [comment]
yes

**Comment by liexes** [comment]
I suggest this because with the freezed, users bug and start shaking, it's annoying, and sometimes it takes them off the server for sending a lot of packets or for flying.

**Comment by RainFallDev** [comment]
yu are changed server version or spigot?

**Comment by liexes** [comment]
I have lightspigot

**Comment by liexes** [comment]
but, this error I've always had it

**Comment by RainFallDev** [comment]
use foxspigot or carbonspigot or windspigot https://github.com/Wind-Development/WindSpigot/releases/tag/v2.0.1

**Comment by liexes** [comment]
What I want is that the player at the beginning, does not freeze him, I want him to let him walk around the map, without being able to hit, until the countdown of the start of the game ends, at the end of the countdown I want him to return the player to a initial position and the match start.

**Comment by RainFallDev** [comment]
@GoodestEnglish

**Comment by RainFallDev** [comment]
@GoodestEnglish

**Comment by liexes** [comment]
The Plugin if it is compatible, has the KB System, with this version, does not present any error, I am just suggesting something

**Comment by RainFallDev** [comment]
Your spigot not supported cant be used use supported spigot

**Comment by RainFallDev** [comment]
the plugin not working on unsupported spigot

**Comment by liexes** [comment]
what? for what? to me the Plugin with lightspigot is perfect, it is as if I had windspigot but improved

**Comment by liexes** [comment]
The plugin works correctly with my lightspigot

**Comment by RainFallDev** [comment]
You can get supported spigots https://github.com/diamond-rip/Eden/wiki/Getting-Started You cant use unsupported spigot

**Comment by RainFallDev** [comment]
You cant play on lightspigot there is no bugs change your spigot version

**Comment by RainFallDev** [comment]
yes the plugin works you can create arenas kits but you cant play in unsupported spigot

**Comment by liexes** [comment]
What do you men with you can't play in unsupported spigot? I can create party, I can play normally, I can send duels, create arenas, I can do everything and do not jump errors in console, everything is normal

**Comment by RainFallDev** [comment]
yes but after play game you have freeze this is spigot bug

**Comment by GoodestEnglish** [comment]
The purpose of "supported spigot forks" just means Eden can hook into their knockback system or movement handler system. Other spigot forks are welcome but not guarantee to work.

You should be able to move around the map before the match starts. Do you have `start freeze` in kit game rule disabled?

**Comment by liexes** [comment]
No @RainFallDev Don't freeze me, I say I don't want you to freeze the player at the start of the game

**Comment by liexes** [comment]
I have it activated @GoodestEnglish Because if I leave it off, yes, they can be moved, but it does not return them to the bone starting point, position A and B

**Comment by GoodestEnglish** [comment]
So this is basically a enhancement. But unfortunately due to the flood of enhancement, I've currently not accepting any enhancements.

(My personal opinion but I think it is a bit awkward to do that for a practice server. I've never seen any servers do that)

**Comment by liexes** [comment]
Most Latin American practice servers have this system, since freezing bothers

**Comment by liexes** [comment]
But if then you can't implement it, would it be possible to get some placeholders? that's how I do it with the Plugin conditional events

@GoodestEnglish

**Comment by liexes** [comment]
I would need rI would need a placeholder what of the position a and b of the arena in which the player is, one that gives me the name of the kit and another that gives me the countdown, If you can both, better, example:

1. %eden_arenaspawn_<a/b>
2. %eden_arenaname%
3. %eden_countdown_<start/respawn>%

**Comment by HierarchyMan** [comment]
> I have lightspigot

Keep using it, fox has broken events

**Comment by HierarchyMan** [comment]
> So this is basically a enhancement. But unfortunately due to the flood of enhancement, I've currently not accepting any enhancements.
> 
> (My personal opinion but I think it is a bit awkward to do that for a practice server. I've never seen any servers do that)

Pika and hypixel do that but they don't count cuz both trash

**Comment by liexes** [comment]
> I would need rI would need a placeholder what of the position a and b of the arena in which the player is, one that gives me the name of the kit and another that gives me the countdown, If you can both, better, example:
> 
> 1. %eden_arenaspawn_<a/b>%
> 2. %eden_arenaname%
> 3. %eden_countdown_<start/respawn>%

@GoodestEnglish ?

**Comment by RainFallDev** [comment]
Scoreboard placeholder
As the name said, scoreboard placeholders are placeholder which are for scoreboard only. Here's all the placeholders:

{event-information} - The current event's information. Event's information message is editable via language.yml (event.tournament.scoreboard)
{party-leader}
{party-members}
{party-max}
{event-uncolored-name}
{event-total-players}
{event-max-players}
{event-countdown}
{queue-kit}
{queue-time}
{queue-ranked-min}
{queue-ranked-max}
{ping-range}
{match-kit}
{match-duration}
{match-build-limit}
{match-build-limit-difference}
{match-team<team number>-logo}
{match-team<team number>-bed-status}
{match-team<team number>-points}
{match-solo-opponent}
{match-solo-winner}
{match-solo-loser}
{match-solo-boxing-difference-text}
{match-solo-boxing-difference}
{match-solo-boxing-difference-number}
{match-solo-boxing-difference-symbol}
{match-solo-boxing-difference-color}
{match-solo-boxing-self-hit}
{match-solo-boxing-opponent-hit}
{match-solo-boxing-combo}
{match-solo-self-ping}
{match-solo-opponent-ping}
{match-team-self-alive}
{match-team-self-size}
{match-team-opponent-alive}
{match-team-opponent-size}
{match-team-winner}
{match-team-loser}
{match-team-boxing-difference-text}
{match-team-boxing-difference}
{match-team-boxing-difference-symbol}
{match-team-boxing-difference-color}
{match-team-boxing-self-hit}
{match-team-boxing-opponent-hit}
{match-team-boxing-combo}
{match-ffa-alive}
{match-ffa-player-size}
{match-ffa-winner}
{match-ffa-loser}
{spectate-kit}
{spectate-duration}
{spectate-build-limit}
{spectate-build-limit-difference}
{spectate-team<team number>-logo}
{spectate-team<team number>-bed-status}
{spectate-team<team number>-points}
{spectate-solo-player1}
{spectate-solo-player2}
{spectate-solo-winner}
{spectate-solo-loser}
{spectate-solo-boxing-player1-hit}
{spectate-solo-boxing-player2-hit}
{spectate-solo-boxing-player1-combo}
{spectate-solo-boxing-player2-combo}
{spectate-solo-player1-ping}
{spectate-solo-player2-ping}
{spectate-team1-leader}
{spectate-team2-leader}
{spectate-team1-alive}
{spectate-team2-alive}
{spectate-team1-size}
{spectate-team2-size}
{spectate-team-winner}
{spectate-team-loser}
{spectate-team1-boxing-hit}
{spectate-team2-boxing-hit}
{spectate-team1-boxing-combo}
{spectate-team2-boxing-combo}
{spectate-ffa-alive}
{spectate-ffa-player-size}
{spectate-ffa-winner}
{spectate-ffa-loser}
{online-players}
{queue-players}
{match-players}
PlaceholderAPI's placeholder
PlaceholderAPI's placeholder are placeholders which handles by PlaceholderAPI plugin. Which means these placeholder can be access by any plugins which supports PlaceholderAPI. Note that language.yml supports PlaceholderAPI too. You can use external placeholders in language.yml. Here's the placeholders:

%eden_queue_unranked_<kit name>% - Return the unranked queue size of kit
%eden_queue_ranked_<kit_name>% - Return the ranked queue size of kit
%eden_match_unranked_<kit name>% - Return the unranked match size of kit
%eden_match_ranked_<kit name>% - Return the ranked match size of kit
%eden_kit_status_<kit name>% - Return if the kit is enabled or not
%eden_party_privacy% - Return the privacy if the current party. Will return nothing if the player has no party.
%eden_player_ranked_win%
%eden_player_ranked_loss%
%eden_player_unranked_win%
%eden_player_unranked_loss%
%eden_player_overall_win%
%eden_player_overall_loss%
%eden_player_total_elo% - Return the total elo of the player
%eden_player_global_elo% - Return the average elo of the player (total elo / kits amount which is enabled and has ranked mode)
%eden_player_elo_<kit name>%
%eden_player_peakElo_<kit name>%
%eden_player_unrankedWon_<kit name>%
%eden_player_unrankedLost_<kit name>%
%eden_player_rankedWon_<kit name>%
%eden_player_rankedLost_<kit name>%
%eden_player_bestWinstreak_<kit name>%
%eden_player_winstreak_<kit name>%
%eden_leaderboard_bestWinstreak_player_<kit name>_<position>% - Return the player's name of best winstreak leaderboard with and (limit to top 10)
%eden_leaderboard_bestWinstreak_winstreak_<kit name>_<position>% - Return the amount of winstreak of best winstreak leaderboard with and (limit to top 10)
%eden_leaderboard_elo_player_<kit name>_<position>% - Return the player's name of elo leaderboard with and (limit to top 10)
%eden_leaderboard_elo_elo_<kit name>_<position>% - Return the elo of elo leaderboard with and (limit to top 10)
%eden_leaderboard_wins_player_<kit name>_<position>% - Return the player's name of wins leaderboard with and (limit to top 10)
%eden_leaderboard_wins_win_<kit name>_<position>% - Return the amount of wins of wins leaderboard with and (limit to top 10)
%eden_leaderboard_winstreak_player_<kit name>_<position>% - Return the player's name of winstreak leaderboard with and (limit to top 10)
%eden_leaderboard_winstreak_winstreak_<kit name>_<position>% - Return the amount of winstreak of winstreak leaderboard with and (limit to top 10)

**Comment by liexes** [comment]
@RainFallDev I know it's a placeholder bro, only the ones I'm asking for aren't

**Comment by liexes** [comment]
I correct, the ones I need

**Comment by RainFallDev** [comment]
ok

**Comment by liexes** [comment]
> I would need rI would need a placeholder what of the position a and b of the arena in which the player is, one that gives me the name of the kit and another that gives me the countdown, If you can both, better, example:
> 
> 1. %eden_arenaspawn_<a/b>%
> 2. %eden_arenaname%
> 3. %eden_countdown_<start/respawn>%

Bro @GoodestEnglish ?

**Comment by GoodestEnglish** [comment]
I have to know what placeholder you want. Yea I can do arena name but arena spawn is a location, why do you need to display it as a string?

**Comment by liexes** [comment]
It is enclosed in <> the options of each placeholder example %eden_position% can give either the spawn position A or B, but that is already changed in the placeholder example %eden_position_a% gives the spawn position A of the arena in which there is the player

Another example:

The placeholder %eden_countdown% can give 2 things or the start countdown or the respawn countdown.

To give the reappearance it would be like this more or less
%eden_countdown_respawn%


With these placeholders I could do many things and very easily.

Example, with the %eden_countdown_spawn% (This gives the countdown at the start of the game, what was put in the editor kit)
I can make it so that when this placeholder of the variable 0, that is, the countdown is at 0, the player will tepe to his position, b or a. Do you understand me?

**Comment by liexes** [comment]
I would need all of them, to make a box system for the Bridge, a system to teleport the player back to their starting position at the end of the starting content and much more!

PS: I would also need a placeholder that gives me the team of the player in the arena, for example %eden_equip% this will give the team, the message that is in the message.yml for example "Red" or "Blue" If you add these placeholders , I can make several systems for you without having to update the Plugin, for example the box system in the Bridge.

**Comment by liexes** [comment]
In summary, I would need these placeholders in <> I add the possible options is that of this

%eden_arenaspawn_<a/b>%
%eden_arena_name%
%eden_countdown_<start/respawn>%
%eden_arena_equip% (If it is not in a split or duo mode such as bedwars it can give the teams (A and B)

@GoodestEnglish

**Comment by GoodestEnglish** [comment]
> It is enclosed in <> the options of each placeholder example %eden_position% can give either the spawn position A or B, but that is already changed in the placeholder example %eden_position_a% gives the spawn position A of the arena in which there is the player
> 
> Another example:
> 
> The placeholder %eden_countdown% can give 2 things or the start countdown or the respawn countdown.
> 
> To give the reappearance it would be like this more or less %eden_countdown_respawn%
> 
> With these placeholders I could do many things and very easily.
> 
> Example, with the %eden_countdown_spawn% (This gives the countdown at the start of the game, what was put in the editor kit) I can make it so that when this placeholder of the variable 0, that is, the countdown is at 0, the player will tepe to his position, b or a. Do you understand me?

I don't understand. Placeholder is a string. Location is a location. Both are different. If you wish to do extra things with Eden you should modify the Match class/field and listen to MatchStartEvent.

**Comment by liexes** [comment]
> 



> > Está encerrado entre <> las opciones de cada ejemplo de marcador de posición %eden_position% puede dar la posición de aparición A o B, pero eso ya está cambiado en el ejemplo de marcador de posición %eden_position_a% da la posición de aparición A de la arena en la que esta el jugador
> > Otro ejemplo:
> > El marcador de posición %eden_countdown% puede dar 2 cosas o la cuenta regresiva de inicio o la cuenta regresiva de reaparición.
> > Para dar la reaparicion seria asi mas o menos %eden_countdown_respawn%
> > Con marcadores de posicion podria hacer muchas cosas estas y muy facilmente.
> > Ejemplo, con %eden_countdown_spawn% (Esto da la cuenta regresiva al comienzo del juego, lo que se puso en el kit del editor) puedo hacer que cuando este marcador de posición de la variable 0, es decir, la cuenta regresiva esté en 0, el jugador tepe a su posición, bo a. ¿Me entiendes?
> 
> No entiendo. El marcador de posición es una cadena. La ubicación es una ubicación. Ambos son diferentes. Si desea hacer cosas adicionales con Eden, debe modificar la clase/campo Match y escuchar MatchStartEvent.

I just want to add placeholders so that it throws me a certain variable, with which I can play with conditional events, I don't know about java, so I can't add a custom event

**Comment by liexes** [comment]
> > It is enclosed in <> the options of each placeholder example %eden_position% can give either the spawn position A or B, but that is already changed in the placeholder example %eden_position_a% gives the spawn position A of the arena in which there is the player
> > Another example:
> > The placeholder %eden_countdown% can give 2 things or the start countdown or the respawn countdown.
> > To give the reappearance it would be like this more or less %eden_countdown_respawn%
> > With these placeholders I could do many things and very easily.
> > Example, with the %eden_countdown_spawn% (This gives the countdown at the start of the game, what was put in the editor kit) I can make it so that when this placeholder of the variable 0, that is, the countdown is at 0, the player will tepe to his position, b or a. Do you understand me?
> 
> I don't understand. Placeholder is a string. Location is a location. Both are different. If you wish to do extra things with Eden you should modify the Match class/field and listen to MatchStartEvent.

I only want to add placeholders so that it throws or gives me a certain variable, with which I can play with conditional events, I don't know about java, so I can't add a custom raised event when I mean that it throws a certain variable, it gives me a certain message , for example let's take the %eden_party_status% this, if you put it in a /papi parse it will give you something that I call a variable in this case, if the player is not in a party, it will give you the variable no, or if it is in one, it will tell you if it is open, that is exactly what I want with the placeholders that I sent you, for the %eden_arenaspawn% there are 2 different ones, what can you add,
%eden_arenaspawn_a% and %eden_arenaspawn_b% These placeholders can give me position A or B, as I decide, of the arena in which the player is currently playing, for example, if I am playing in a bedwars arena and I put the placeholder %eden_position_a% will give me the position a of the arena, which is placed in the arena creation menu, where the player appears, this will return the coordinates of its location, an example "100 70 100"

**Comment by liexes** [comment]
I will leave you a list that you could throw each of the placeholders that I left you

%eden_countdown_start% the same for %eden_countdown_respawn% Gives the kit countdown at the start of the game, this is set in the kit editing menu, for example by default it is at (5) and if the kit creator modifies it, then to walk as assigned (5, 4, 3, 2, 1, 0, -1) I can make when the counter reaches 0 with this placeholder create certain events, create a box at position x, return the player to the x position, send a title to the player, etc. (It is necessary to add the -1 or when reaching 0 it will be executing this action all the time, the 0 can also be changed to "Started" or "Playing"

%eden_arena_name% Returns the name of the arena the player is in, the name is the one imposed by the creator of the arena, for example "Rubiks" if the player is not in an arena then it could return "None"


%eden_kit_name% This gives the name of the kit in which the user is, for example if he is in bedwars it throws the message or in my case variable, "Bedwars" If he is not in any kit playing, it could give "None" or " not playing"

%eden_equip% This will return the player's equipment in an arena, if it is in an arena like bedwars which is custom then it might return the messages set in the message.yml example "Red and blue".

What can I do with eden placeholders and conditional events?

I have done multiple things so far, I have made a random system to enter a random game of sumo, or etc (I have it by sections) I have created a waiting room, I have created an effects system at the end of the game (This thanks to the latest command update) And many more things.

**Comment by liexes** [comment]
Excuse the mess, the indentation that I imposed in the text, the translator removes it when turning it into English since I speak Spanish.

**Comment by liexes** [comment]
@GoodestEnglish

**Comment by liexes** [comment]
It's just a matter of playing with the variables and actions that the plugin gives, for me it's easy

**Comment by liexes** [comment]
@GoodestEnglish ¿Podrá agregar los placeholders?

**Comment by liexes** [comment]
To be more explanatory, I will create a repetitive event that will detect every 20 seconds each variable that will give the placeholders what I am asking in this case would be %player_world% == arenas when this variable is completed it will continue with another, %eden_kit_name% = = thebridge will then pass to another variable %eden_countdown_start% == ...(Here I need you to give me a variable before starting the count, so I can put the box before the user enters so it doesn't get left out) when all variables are completed and that are equal to those imposed, actions will be executed, also thanks to the placeholders:

- 'console_command: //pos1 %eden_position_a%'
- 'console_command: //pos2 %eden_position_b%'
- 'console_command: schema load cage'
- 'console_command: //paste -a'
- 'wait: 3' (The seconds of the countdown)
- 'console_command: //set 0'

This is just an example of how it could be done, but I won't do it like this because if several users enter the same kit it will be bugged, so I have to make the user execute the command.

**Comment by liexes** [comment]
I can also do modes like tntrun, bowspleef with double jump and triple arrows

**Comment by liexes** [comment]
@GoodestEnglish You could also make a system for making boxes in Skywars, but you would have to make the inventories of the chests keep, because when you take them out once, they no longer appear in other matches.

**Comment by liexes** [comment]
@GoodestEnglish ?

**Comment by GoodestEnglish** [comment]
No. Please always remember, Eden is a community based plugin. Not a plugin which is exclusive for you. 

Features which are exclusive should be done by developer on your side. Not me.

**Comment by liexes** [comment]
I am only asking for some placeholders that from my point of view are the basics that all practice plugins have, just please, I ask you to add them, what I plan to publish the systems so that people who like my systems implement it on their server.

**Comment by liexes** [comment]
Seriously, I can do great things, it's just adding it and if someone asks you for example a suggestion and you do not want to add it, then I can add it easily and that person will be comfortable with the Plugin.

**Comment by liexes** [comment]
Solo es una forma de personalizar, yo le ayudaré en caso de que no quiera agregar algo al plugin, usted solo me dice y lo agrego en menos de 3h, sistemas complejos, puedo enseñarle todo lo que puedo hacer con este Plugin. Se lo pido, por favor, no se java y me va a resultar tardado porque debo aprenderlo.

**Comment by liexes** [comment]
Just add those simple placeholders, please @GoodestEnglish

**Comment by liexes** [comment]
Sorry if I said, you could also add a box system for Skywars, by that I meant that I, I can do it, if not that the translator put it wrong

**Comment by liexes** [comment]
With those placeholders I can do the following things with the Plugin conditional events:

- Tntrun mode with 2 jump
- Bowspleef with douvle jump and triple arrow
- cages for skywars and other mode
- parkour (I already did)
- Portal goal ffa mode
- Walk freely at the beginning of the game, at the end of the count will be returned to the spawn
- tnttag mode


I have some systems, that's what I call them, already created, but they have nothing to do, it's just to demonstrate my ability:

- Crates system (100% custom)
- Ranked and unranked match random
- A custom waiting lobby when a player is in queue

And more (I can send proofs)

It's just that you tell users that I can make custom systems for your plugin whatever they want, making the plugin more striking and complete.

**Comment by liexes** [comment]
@HierarchyMan You can considere this? (sorry for the ping)

**Comment by HierarchyMan** [comment]
> @HierarchyMan You can considere this? (sorry for the ping)

Not really but since Eden isn't ripping off MMC like other paid practice plugins, a feature like this would make it stand apart
I believe this feature, if ever added should be disabled by default.
COULD be counted as enhancement
And as Goodest English stated he is currently only working on big fixes and core features for now

**Comment by liexes** [comment]
Ho... Sorry, the mensaje no has for you, @GoodestEnglish please, is a simple placeholders :(

**Comment by liexes** [comment]
> With those placeholders I can do the following things with the Plugin conditional events:
> 
> * Tntrun mode with 2 jump
> * Bowspleef with douvle jump and triple arrow
> * cages for skywars and other mode
> * parkour (I already did)
> * Portal goal ffa mode
> * Walk freely at the beginning of the game, at the end of the count will be returned to the spawn
> * tnttag mode
> 
> I have some systems, that's what I call them, already created, but they have nothing to do, it's just to demonstrate my ability:
> 
> * Crates system (100% custom)
> * Ranked and unranked match random
> * A custom waiting lobby when a player is in queue
> 
> And more (I can send proofs)
> 
> It's just that you tell users that I can make custom systems for your plugin whatever they want, making the plugin more striking and complete.

Read this please @GoodestEnglish

**Comment by liexes** [comment]
> Ho... Sorry, the mensagge no has for you, @GoodestEnglish please, is a simple placeholders :(

**Comment by liexes** [comment]
From my point of view it is not an improvement, it is only a position marking itself, all, most practice plugins have it, this serves to place different things, such as the kit on the scoreboard and more, I just ask you to add them please

**Comment by Chromaah** [comment]
> So this is basically a enhancement. But unfortunately due to the flood of enhancement, I've currently not accepting any enhancements.
> 
> (My personal opinion but I think it is a bit awkward to do that for a practice server. I've never seen any servers do that)

Hypixel does this
